NewHorizons/Builder/Props/WarpPadBuilder.cs

```diff
@@ -0,0 +1,149 @@
+
+                var lamp = thReceiverLamp.InstantiateInactive();
+                lamp.transform.parent = _detailedReceiverPrefab.transform;
+                lamp.transform.localPosition = thReceiver.transform.InverseTransformPoint(thReceiverLamp.transform.position);
+                lamp.transform.localRotation = thReceiver.transform.InverseTransformRotation(thReceiverLamp.transform.rotation);
+
+                _detailedReceiverPrefab.SetActive(false);
+                lamp.SetActive(true);
+                detailedReceiver.SetActive(true);
+
+                _detailedReceiverPrefab.DontDestroyOnLoad();
+
+                GameObject.Destroy(_detailedReceiverPrefab.GetComponentInChildren<NomaiWarpStreaming>().gameObject);
+            }
+
+            if (_receiverPrefab == null)
+            {
+                _receiverPrefab = SearchUtilities.Find("SunStation_Body/Sector_SunStation/Sector_WarpModule/Interactables_WarpModule/Prefab_NOM_WarpReceiver")
+                    .InstantiateInactive()
+                    .DontDestroyOnLoad();
+                GameObject.Destroy(_receiverPrefab.GetComponentInChildren<NomaiWarpStreaming>().gameObject);
+
+                var structure = _platformContainerPrefab.Instantiate();
+                structure.transform.parent = _receiverPrefab.transform;
+                structure.transform.localPosition = new Vector3(0, 0.8945f, 0);
+                structure.transform.localRotation = Quaternion.identity;
+                structure.SetActive(true);
+            }
+
+            if (_transmitterPrefab == null)
+            {
+                _transmitterPrefab = SearchUtilities.Find("TowerTwin_Body/Sector_TowerTwin/Sector_Tower_SS/Interactables_Tower_SS/Tower_SS_VisibleFrom_TowerTwin/Prefab_NOM_WarpTransmitter")
+                    .InstantiateInactive()
+                    .DontDestroyOnLoad();
+                GameObject.Destroy(_transmitterPrefab.GetComponentInChildren<NomaiWarpStreaming>().gameObject);
+
+                var structure = _platformContainerPrefab.Instantiate();
+                structure.transform.parent = _transmitterPrefab.transform;
+                structure.transform.localPosition = new Vector3(0, 0.8945f, 0);
+                structure.transform.localRotation = Quaternion.identity;
+                structure.SetActive(true);
+            }
+        }
+
+        public static void Make(GameObject planetGO, Sector sector, NomaiWarpReceiverInfo info)
+        {
+            var receiverObject = GeneralPropBuilder.MakeFromPrefab(info.detailed ? _detailedReceiverPrefab : _receiverPrefab, "NomaiWarpReceiver", planetGO, sector, info);
+
+            StreamingHandler.SetUpStreaming(receiverObject, sector);
+
+            var receiver = receiverObject.GetComponentInChildren<NomaiWarpReceiver>();
+
+            receiver._frequency = GetFrequency(info.frequency);
+
+            receiver._alignmentTarget = planetGO?.transform;
+
+            Locator.RegisterWarpReceiver(receiver);
```

Comment:
receiver already registers itself in Awake so i dont think this is needed?

Reply:
You're god damn right